docs/reference/src/main/asciidoc/ri-spi.asciidoc

```diff
@@ -979,3 +979,101 @@ Instead, this dependency needs to be deployed separately to the OSGi container.
+It also allows to fire both kinds of events; synchronous and asynchronous.
+See the example below:
+
+===== Weld-core artifact change 
+
+For historical reasons, Weld 2.x core is released in two versions which *are identical* in their contents.
+Those are `org.jboss.weld:weld-core` and `org.jboss.weld:weld-core-impl`.
+Starting with Weld 3.0, only `org.jboss.weld:weld-core-impl` artifact will be released.
+
+[source.JAVA, java]
+--------------------------------------------------------------------------------------------------
+@Inject
+BeanManager bm;
+
+public void fireMyEvent() {
+    bm.getEvent().select(Any.Literal.INSTANCE).fire(new Payload());
+    bm.getEvent().select(PaymentCheck.class, Default.Literal.INSTANCE)
+        .fireAsync(new Payload());
+}
+--------------------------------------------------------------------------------------------------
+
+===== SPI changes
+
+The SPI changes are mainly important for integrators, such as WildFly.
+Here is a list of changes based on what SPI class/method they affect:
+
+* http://docs.jboss.org/weld/javadoc/3.0/weld-spi/org/jboss/weld/bootstrap/spi/BeanDeploymentArchive.html[`org.jboss.weld.bootstrap.spi.BeanDeploymentArchive`]
+
+** Two new methods - `getKnownClasses()` and `getLoadedBeanClasses()`
+** Both methods have default implementations
+** Integrators might want to override them if they want to make use of preloaded classes during bootstrap
+** For detailed info, take a peek at Javadoc of these methods
+
+* http://docs.jboss.org/weld/javadoc/3.0/weld-spi/org/jboss/weld/bootstrap/spi/BeansXml.html[`org.jboss.weld.bootstrap.spi.BeansXml`]
+
+** One new method - `isTrimmed()`
+** As per https://issues.jboss.org/browse/CDI-420[CDI-420], this reflects the altered bean discovery mode
+** If an integrator provides their own implementation of `BeansXml` interface, they will now need to implement this method as well
+
+* http://docs.jboss.org/weld/javadoc/3.0/weld-spi/org/jboss/weld/security/spi/SecurityServices.html[`org.jboss.weld.security.spi.SecurityServices`]
+
+** One new method - `getSecurityContext()`
+** Returns current security context, which can then be propagated to different threads
+** This method has a default implementation which return a no-op implementation
+
+* http://docs.jboss.org/weld/javadoc/3.0/weld-spi/org/jboss/weld/security/spi/SecurityContext.html[`org.jboss.weld.security.spi.SecurityContext`]
+
+** Newly added interface which Weld uses to associate and dissociate security context with given thread
+** The main use case for this are asynchronous event observers which are executed in separate threads, yet need to have identical context
+** There are three methods in this interface - `associate()`, `dissociate()` and `close()`
+
+* http://docs.jboss.org/weld/javadoc/3.0/weld-spi/org/jboss/weld/manager/api/WeldManager.html[`org.jboss.weld.manager.api.WeldManager`]
+
+** Following methods were removed from the SPI:
+*** `WeldManager.createActivity() : WeldManager`
```

Comment:
A return type of the method.
That is a syntax for OOP design I recall from university.
Was something like: `myMethod(Param) : ReturnType`
But I can change this if you don't like it.